packages/builder/src/components/design/settings/controls/ButtonActionEditor/actions/DownloadFile.svelte

```diff
@@ -0,0 +1,76 @@
+<script>
+  import { Select, Label } from "@budibase/bbui"
+  import { onMount } from "svelte"
+  import DrawerBindableInput from "components/common/bindings/DrawerBindableInput.svelte"
+  import { FieldType } from "@budibase/types"
+
+  export let parameters
+  export let bindings = []
+
+  $: fileBindings = bindings?.filter(
+    b => b.fieldSchema?.type === FieldType.ATTACHMENT
+  )
+
+  const fileOptions = [
+    {
+      label: "Attachment",
+      value: "attachment",
+    },
+    {
+      label: "Url",
+      value: "url",
+    },
+  ]
+
+  onMount(() => {
+    if (!parameters.type) {
+      parameters.type = "attachment"
+    }
+  })
+</script>
+
+<div class="root">
+  <Label small>File</Label>
+  <Select
+    placeholder={null}
+    bind:value={parameters.type}
+    options={fileOptions}
+  />
+  {#if parameters.type === "attachment"}
+    <Label small>Attachment</Label>
+    <DrawerBindableInput
+      title="Attachment"
+      bindings={fileBindings}
+      allowHelpers={false}
+      value={parameters.attachment}
+      on:change={value => (parameters.attachment = value.detail)}
+    />
```

Comment:
How will this work, for multiple attachment columns?

Reply:
I guess how we could make it work is if you pick an attachment column with multiple then we just download them all. This would work well for old attachment columns with just a single attachment.

Might be a bit weird though if you have like 10 different files attached, we're going to trigger 10 downloads - but I think this makes some sense and if you just want to download one of the files then you use the URL + name to pick a specific one.

Reply:
@mike12345567 yep I agree - this is what @adrinr and I discussed offline and decided on yesterday as well, so great minds think alike! 👌

Reply:
Done! If there is more than 1 file in the column, it gets zipped!

Reply:
Just for clarity - this has ended up looking like:
![image](https://github.com/Budibase/budibase/assets/4407001/ff96693f-8954-4ed4-a627-373938ebb850)

The zip name is based on the display column of the row.